### PR TITLE
fix(element): return not present when an element disappears

### PR DESCRIPTION
**Pull request by hankduan**

### CODE REVIEW

File: lib/element.js [hunk 1]
```diff
@@ -353,7 +353,29 @@ ElementArrayFinder.prototype.toElementFinder_ = function() {
  */
 ElementArrayFinder.prototype.count = function() {
   return this.getWebElements().then(function(arr) {
-    return arr.length;
+    var list = [];
+    arr.forEach(function(webElem) {
```
Comment (tullmann) [inline lib/element.js]: A `forEach` that ends with a list `push` is just a `map` call?   Structuring this as `list = arr.map(function(webElem) { ... })` might be cleaner.

Comment (hankduan) [inline lib/element.js]: Changing to use map + reduce instead.

**Comment by sjelin** [comment]
LGTM

**Comment by hankduan** [comment]
committed as https://github.com/angular/protractor/commit/2a765c76e121de13ff86a279fe3f21cb15c17783

**Pull request closed without merging**